src/Common.OData.ApiExplorer/AspNet.OData/DefaultModelTypeBuilder.cs

```diff
@@ -66,6 +67,14 @@ public Type NewStructuredType( IEdmStructuredType structuredType, Type clrType,
                 var propertyType = property.PropertyType;
                 var structuredTypeRef = structuralProperty.Type;
 
+                if ( passedTypes.TryGetValue( structuredTypeRef.ToString(), out var passedType ) )
```

Comment:
Let's capture the type name so we don't have to build it two times. Something like:

```c#
var typeName = structuredTypeRef.FullName();
```

Reply:
This obviously supplanted by the struct key or hash code if you refactor